helm/cloud-provider-vsphere/templates/secret.yaml

```
@@ -1,15 +1,15 @@
-{{- if .Values.config.enabled | default .Values.global.config.enabled -}}
+{{- if and .Values.config.secret.create .Values.config.enabled | default .Values.global.config.enabled -}}
```

```suggestion
{{- if and .Values.config.secret.create .Values.config.enabled -}}
```
There are already default values in `values.yaml`.

---

I agree, however thats the upstream impl: https://github.com/kubernetes/cloud-provider-vsphere/blob/master/charts/vsphere-cpi/templates/secret.yaml#L1 Do we want to diverge? This comes with a cost. Currently the sync was only `"cp -r foo bar"` + adding those `{{- include "labels.common" $ | nindent 4 }}`

---

Ah right, weird... yeah let's not diverge then.